packages/world/src/Utils.sol

```diff
@@ -5,17 +5,24 @@ import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
 import { ResourceId, WorldResourceIdInstance } from "./WorldResourceId.sol";
 import { SystemRegistry } from "./codegen/tables/SystemRegistry.sol";
 
+/**
+ * @notice Various utilities
+ * @dev These utilities are not used by MUD itself, they are for developers using MUD.
+ */
 library Utils {
   using WorldResourceIdInstance for ResourceId;
 
   /**
-   * Get the namespace of this system.
-   * Must be used within the context of a system (either directly, or within libraries called by a system).
-   *
-   * Note unlike systemNamespace, getting systemName is impossible for root systems,
-   * because they're delegatecalled and the name isn't passed in calldata
+   * @notice Fetches the namespace of the current system.
+   * @dev This function determines the system's namespace based on its interaction with the store. If the system is a root system (direct call), it returns an empty string. Otherwise, it retrieves the namespace using the system's registry.
```

Comment:
```suggestion
   * @dev This function determines the system's namespace based on its interaction with the store. If the system is a root system, it returns the root namespace (an empty string). Otherwise, it retrieves the namespace using the system's registry.
```